internal/middleware/usagemetrics/usagemetrics.go

```diff
@@ -0,0 +1,138 @@
+package usagemetrics
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors"
+	prometheus "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/rs/zerolog/log"
+	"google.golang.org/grpc"
+
+	dispatch "github.com/authzed/spicedb/internal/proto/dispatch/v1"
+	"github.com/authzed/spicedb/pkg/responsemeta"
+)
+
+var dispatchBuckets = []float64{1, 5, 10, 25, 50, 100, 250}
+
+var dispatchedCountHistogram = promauto.NewHistogramVec(prometheus.HistogramOpts{
+	Namespace: "spicedb",
```

Comment:
If this is made generic/importable, do these values make sense? should they be pluggable? is it enough for the instantiator to be able to supply another label value?